### PR TITLE
Remove tfs-impl-10

### PR DESCRIPTION
**Pull request by varshavaradarajan**
@arvindsv , @ketan mentioned you had some concerns around this.

### CODE REVIEW

File: server/test/unit/com/thoughtworks/go/server/controller/AgentRegistrationControllerTest.java [hunk 1]
```diff
@@ -65,7 +65,6 @@ public void setUp() throws Exception {
         goConfigService = mock(GoConfigService.class);
 
         when(systemEnvironment.getSslServerPort()).thenReturn(8443);
-        when(systemEnvironment.get(SystemEnvironment.TFS_SDK_10)).thenReturn(true);
```
Comment (jyotisingh) [inline server/test/unit/com/thoughtworks/go/server/controller/AgentRegistrationControllerTest.java]: Won't removing that expectation break any of the tests? May be trigger a PR build, but also run the tests locally for quick feedback

Comment (varshavaradarajan) [inline server/test/unit/com/thoughtworks/go/server/controller/AgentRegistrationControllerTest.java]: Removed the checks for the system property. Will keep an eye on the PR build.

**Comment by jyotisingh** [comment]
@varshavaradarajan - PR looks good, will merge after regression-tests passes.
Pending: https://github.com/gocd/docs.go.cd/issues/99
As far as I remember, we did not announce the JVM arg that is being removed as a part of this PR, but double check if the documentation for the same.

**Comment by varshavaradarajan** [comment]
Regression passed, @jyotisingh. Will make the doc changes.

**Pull request merged**